src/components/RightSidebar/Participants/Participant.vue

```diff
@@ -787,7 +788,16 @@ export default {
 			if (!value || !(value === 'ringing' || value === 'accepted')) {
 				this.disabled = false
 			}
-		}
+		},
+
+		participantNavigationId() {
+			// FIXME: as RecycleScroller reuses item components, we need to reset the state,
+			// if participant object has been replaced by another one
+			this.permissionsEditor = false
+			this.isRemoveDialogOpen = false
+			this.isBanParticipant = false
+			this.internalNote = ''
+		},
```

Comment:
This is expected behavior of `RecycleScroller` and the same may happen in other cases where the component is reused, but it doesn't fully support props update.

This `watch` fixes props update support.

But in general case I'd:
- Add `:key` where we render this component in `ParticipantsListVirtual`
- Remove `:key` from `<NcListItem>` in this component (not needed)

It would be the same as setting `:key` with `v-for`. With `key` we don't need to worry about missing one property in this watch or similar issues N components deeper in the tree.

Reply:
> Add :key where we render this component in ParticipantsListVirtual

Isn't that killing the meaning of RecycleScroller? But I'll try with large lists

Reply:
The meaning of the virtual list is to render only visible items and not all items. So if there are 500 participants in the call, only 20 are rendered in the document.

Adding `:key` doesn't lead to rendering all items.

It makes scrolling slower by re-creating a node inside the element. But the same does existing `:key` on `NcListItem` on the next child.

The best (most performant) case is when all the components in the virtual list are stateless, having only props and data from outside the components, but not local data. Making it may require more changes.